pkgs/dartpad_ui/lib/widgets.dart

```diff
@@ -270,3 +287,493 @@ final class Logo extends StatelessWidget {
+            side: BorderSide(color: theme.colorScheme.outline),
+          ),
+          title: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Text(widget.title),
+              if (!_done) const CircularProgressIndicator(),
+            ],
+          ),
+          contentTextStyle: theme.textTheme.bodyMedium,
+          contentPadding: const EdgeInsets.fromLTRB(24, defaultSpacing, 24, 8),
+          content: SizedBox(
+            width: 700,
+            child: widget.existingSource == null
+                ? ReadOnlyEditorWidget(_generatedCode.toString())
+                : ReadOnlyDiffWidget(
+                    existingSource: widget.existingSource!,
+                    newSource: _generatedCode.toString(),
+                  ),
+          ),
+          actions: [
+            Row(
+              children: [
+                Expanded(
+                  child: Align(
+                    alignment: Alignment.centerLeft,
+                    child: RichText(
+                      text: TextSpan(
+                        text: 'Powered by ',
+                        style: DefaultTextStyle.of(context).style,
+                        children: [
+                          TextSpan(
+                            text: 'Google AI',
+                            style: TextStyle(color: theme.colorScheme.primary),
+                            recognizer: TapGestureRecognizer()
+                              ..onTap = () {
+                                url_launcher.launchUrl(
+                                  Uri.parse('https://ai.google.dev/'),
+                                );
+                              },
+                          ),
+                          TextSpan(
+                            text: ' and the Gemini API',
+                            style: DefaultTextStyle.of(context).style,
+                          ),
+                        ],
+                      ),
+                    ),
+                  ),
+                ),
+                TextButton(
+                  onPressed: () => Navigator.pop(context),
+                  child: const Text('Cancel'),
+                ),
+                TextButton(
+                  onPressed: _done ? _onAccept : null,
+                  child: Text(
+                    'Accept',
+                    style: TextStyle(
+                      color: !_done ? theme.disabledColor : null,
+                    ),
+                  ),
+                ),
+                TextButton(
+                  onPressed: _done ? _onAcceptAndRun : null,
+                  child: Text(
+                    'Accept & Run',
+                    style: TextStyle(
+                      color: !_done ? theme.disabledColor : null,
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _onAccept() {
+    assert(_done);
```

Comment:
nit: this widget is already checking for _done in the `onPressed` callback - let's take this assert out. You could also consider moving the check here to make sure this will no-op when the prompt is generating.

Reply:
the whole point of the assert is to state my assumptions coming into this function. if I somehow screwed up the onPressed callback, this assert would alert me.

Reply:
I think what I'm wondering is, now that this widget is finished, why not just make this a no-op when _done is false so that no code in the future has to worry about it?

Reply:
this is just the first version that hasn't been vetted yet in the field. I expect to be doing some maintenance, at which point, having dev-time assumptions documented in the code is useful for flagging issues in a noisy, immediate way instead of a sneaky, non-intuitive way. since the asserts are compiled out of the release version of the code, leaving in assertions seem like all upside to me.

anyway, that's been my experience writing s/w over the years. if you prefer me to remove the assertion, I will.